### ISSUE TITLE
Add ability to turn fci-mode off when the indicator is off of the screen

### ISSUE THREAD

**Issue by dakrone** [alpaker/fill-column-indicator#67]
It'd be great if FCI could be turned off if the buffer wasn't wide enough to show it, so that the fringe doesn't show characters to the right all the time.

So when the buffer is wide enough to display it:

![screenshot from 2016-02-29 14-02-07](https://cloud.githubusercontent.com/assets/19060/13408844/2030534e-deed-11e5-9a8f-0ca5fc86eb7a.png)

However, when the buffer is not wide enough, you get all kinds of arrows in the fringe (there's nothing to the right but the fci column though):

![screenshot from 2016-02-29 14-02-30](https://cloud.githubusercontent.com/assets/19060/13408857/2b4b7aba-deed-11e5-8f57-c2f8d31f05f0.png)

It'd be great if FCI just turned itself off when the buffer isn't wide enough, because it doesn't help when entirely hidden

**Comment by dakrone** [comment]
Here's a workaround that works for me:

```emacs-lisp
(use-package fill-column-indicator
  :ensure t
  :init
  (add-hook 'prog-mode-hook #'fci-mode)
  :config
  (defvar eos/fci-disabled nil)
  (make-variable-buffer-local 'eos/fci-disabled)
  ;; Add a hook that disables fci if enabled when the window changes and it
  ;; isn't wide enough to display it.
  (defun eos/maybe-disable-fci ()
    (interactive)
    ;; Disable FCI if necessary
    (when (and fci-mode
               (< (window-width) (or fci-rule-column fill-column)))
      (fci-mode -1)
      (setq-local eos/fci-disabled t))
    ;; Enable FCI if necessary
    (when (and eos/fci-disabled
               (eq fci-mode nil)
               (> (window-width) (or fci-rule-column fill-column)))
      (fci-mode 1)
      (setq-local eos/fci-disabled nil)))

  (defun eos/add-fci-disabling-hook ()
    (interactive)
    (add-hook 'window-configuration-change-hook
              #'eos/maybe-disable-fci))
  (add-hook 'prog-mode-hook #'eos/add-fci-disabling-hook))
```

**Comment by dakrone** [comment]
@alpaker is there interest in this? I notice it's similar to https://github.com/alpaker/Fill-Column-Indicator/issues/44

**Comment by alpaker** [comment]
Just FYI, I'm abandoning this package and welcoming requests to be take over as maintainer.

**Comment by alpaker** [comment]
The functionality provided by this package has now been implemented natively in Emacs as `display-fill-column-indicator-mode`, available as of version 27.0.90. Unless you are forced to use an older Emacs, you should use the native implementation instead of `fci-mode`. The former is unaffected by most of the compatibility issues that this add-on has.
